src/NEPTypes.jl

```diff
@@ -1016,38 +1030,41 @@ Returns true/false if the NEP is sparse (if compute_Mder() returns sparse)
         return z
     end
 
-    function compute_Mlincomb!(
-                        nep::SPMF_NEP,
-                        λ::Number,
-                        V::AbstractVecOrMat,
-                        a::Vector=ones(size(V,2)))
-
+    function compute_Mlincomb!(nep::SPMF_NEP{T,Ftype},
+                               λ::Number,
+                               V::AbstractVecOrMat,
+                               a::Vector=ones(size(V,2))) where {T,Ftype}
 
-        if ndims(V)==2
-            n,k=size(V);
-        else
-            n=size(V,1); k=1;
-        end
+        local n,k;
+        n=size(V,1);
+        k=size(V,2);
 
     	# we need to assume that the elements of a are different than zero.
     	V[:,findall(x->x==0,a)] .= 0
     	a[findall(x->x==0,a)] .= 1
-    	S=diagm(0 => λ*ones(eltype(V),k)) + diagm(-1 => (a[2:k]./a[1:k-1]).*(1:k-1))
-
-        z=zeros(eltype(V),n)
-        if ndims(V)==1
-            for i=1:size(nep.A,1)
-                Fi=nep.fi[i](S);
-                z=z .+ nep.A[i]*(V*Fi);
-            end
+        local S;
+        if (V isa AbstractVector)
+            S=λ # Vector means just compute matrix vector
         else
-            for i=1:size(nep.A,1)
-                Fi=nep.fi[i](S)[:,1];
-                z=z .+ nep.A[i]*(V*Fi);
-            end
-    	end
+            # V matrix means compute linear combination of derivatives. Use
+            # scaling trick
+       	    S=diagm(0 => λ*ones(eltype(λ),k)) + diagm(-1 => (a[2:k]./a[1:k-1]).*(1:k-1))
```

Comment:
Very marginal comment:
should you define `TT ` earlier and then `S` with the correct type? Namely

```
        # Type logic
        Fλtype=promote_type(eltype(λ),Ftype);
        TT=promote_type(Fλtype,eltype(V)); # Return type
     	S=diagm(0 => λ*ones(TT,k) + diagm(-1 => (a[2:k]./a[1:k-1]).*(1:k-1)) 
```
Or at least, when you define `S` has to have the same type of `a`
`        	    S=diagm(0 => λ*ones(eltype(a),k)) + diagm(-1 => (a[2:k]./a[1:k-1]).*(1:k-1)) `
It looks like a good practice to compute the type at the very top of the function.

Reply:
Indeed. It should be precomputed, and it is currently not correct.

Reply:
Fixed.